### DISCUSSION TITLE
What am I doing wrong with setting up this switch

### DISCUSSION THREAD

**Discussion by chadwtkns** [shadcn-ui/ui#3602] [General]
<img width="475" alt="Screenshot 2024-04-25 at 10 21 43 AM" src="https://github.com/shadcn-ui/ui/assets/5611546/ecd933b3-9e55-486f-b0d1-f78eb09615e8">
<img width="475" alt="Screenshot 2024-04-25 at 10 21 47 AM" src="https://github.com/shadcn-ui/ui/assets/5611546/ead587a1-af9e-47bc-80eb-54b332fd7583">

The toggle isnt showing properly, its showing just a line. Im seeing my IDE complain about ESLint not finding the classname too but I don't know if that's related to anything.

<img width="996" alt="Screenshot 2024-04-24 at 4 01 35 PM" src="https://github.com/shadcn-ui/ui/assets/5611546/26a119cc-6516-4711-8cbd-de2479a67a9a">

Im happy to provide other information too if that's helpful but I haven't really done much configuration other than change the CSS property of --primary to the purple you see on the screenshot

**Comment by chadwtkns** [comment]
nevermind, I figured out that the application we were pulling it into was applying some styles I wasn't expecting. Im still seeing the classnames issue but visually it looks good

**Comment by joaodarocha** [comment]
If anyone else has this problem with the Switch, I got it fixed by adding `text-[0px]` to the `SwitchPrimitive.Root`

Before:
![image](https://github.com/user-attachments/assets/84a78062-0351-4311-8c8b-c2047a66e72a)

Change:
![image](https://github.com/user-attachments/assets/9a4fe153-0ac1-4841-bfa2-7c7398683218)


After:
![image](https://github.com/user-attachments/assets/023a9bf6-5385-487c-a06e-5bd6bc71c2e6)


I explain how I did it here:
https://x.com/JoaoRochaOnX/status/1855200403245130136

**Reply by abena07** [reply]
works 🫶🏿
